packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/apis/get.isolated.test.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import {
+  getSavedObjectFromSourceMock,
+  isFoundGetResponseMock,
+  rawDocExistsInNamespaceMock,
+} from './get.isolated.test.mocks';
+import { SavedObject, CheckAuthorizationResult } from '@kbn/core-saved-objects-server';
+import { apiContextMock, ApiExecutionContextMock } from '../../mocks';
+import { performGet } from './get';
+
+const createSavedObject = (id = 'foo'): SavedObject => {
+  return {
+    id,
+    type: 'bar',
+    attributes: {},
+    references: [],
+  };
+};
+
+describe('performGet', () => {
+  let apiContext: ApiExecutionContextMock;
+
+  beforeEach(() => {
+    apiContext = apiContextMock.create({
+      allowedTypes: ['type'],
+    });
```

Comment:
So, we do have unit tests for SOR method, and Tina did an amazing job extracting them to individual files, however those unit tests are "old school", and don't mock (or allow to mock) much (we're even still using the SOR instance to perform the API calls), so it really wasn't going to work for my test needs where I just wanted to assert that `migrateAndDecryptStorageDocument` is properly called. 

This is why I created new `{apiMethod}.isolated.test` files, that use a different approach, call the api function directly and use a mocked API context to more assert underlying calls instead of full behavior of the APIs.

Reply:
>  use a mocked API context to more assert underlying calls

At last we can start improving these tests rather than spending 99% effort in _finding_ them 😅 !